### ISSUE TITLE
This call to matplotlib.use() has no effect because the backend has already been chosen

### ISSUE THREAD

**Issue by Licko0909** [calico/basenji#34]
Dear developer, this tutorial is very good! Now I have some problems. In the tutorial `tutorial/genes.ipynb`, I ran the following line and I got an error. I checked the related issues and I still didn't solve them.

![image](https://user-images.githubusercontent.com/27897166/56290879-b4ce2000-6156-11e9-86a2-ffc1b9d14d56.png)

In addition, the final output  result "R2, etc."  of ''gene.ipynb' also has problems, as follows:

![image](https://user-images.githubusercontent.com/27897166/56291252-86047980-6157-11e9-8c54-7a98b6b66690.png)

Before that, I installed `stat` according to the prompt. I don't know if it will affect. I want to know the version of your `stat` and `matlibplot` modules, can I?

![image](https://user-images.githubusercontent.com/27897166/56291340-b21ffa80-6157-11e9-876b-d74c6cf86475.png)

**Comment by Licko0909** [comment]
However, I did not make a mistake in drawing PDF in `train_test.ipynb`, as follows:

![image](https://user-images.githubusercontent.com/27897166/56291842-9bc66e80-6158-11e9-9cd1-0a50d4d70bc2.png)

So this inspired me, I changed ‘agg’ to ‘pdf’, and finally there was no warning of `matplotlib.use()`, but the output still has problems, as follows:

![image](https://user-images.githubusercontent.com/27897166/56292425-c664f700-6159-11e9-868b-d9e9d56c9a5e.png)

![image](https://user-images.githubusercontent.com/27897166/56292486-db418a80-6159-11e9-8fc3-9c9e4f9d2d9f.png)

I hope I can get a reply! thank you very much!

**Comment by davek44** [comment]
Hi, I removed the stats.py dependency. That was a personal library of mine that isn't available from pip. I ran the tutorial on my end, and everything worked. So maybe pull the latest commit and try again.

**Comment by Licko0909** [comment]
> Hi, I removed the stats.py dependency. That was a personal library of mine that isn't available from pip. I ran the tutorial on my end, and everything worked. So maybe pull the latest commit and try again.

Thank you very much!

Without retrieving the latest basenji, I solved the problem, I used the model you provided and there was no abnormal output.

![image](https://user-images.githubusercontent.com/27897166/56347392-c49d4100-61f6-11e9-950d-a70429581c3e.png)


This also shows that there have some problem with the training process and the `train.tfr` file obtained by the preprocessing. So I listened to your suggestion, pulled the latest basenji, started a new attempt, and I will re-question.
